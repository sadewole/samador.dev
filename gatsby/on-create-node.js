const { createFilePath } = require(`gatsby-source-filesystem`)

const onCreateNode = ({ node, actions, getNode }) => {
  const { createNodeField } = actions

  if (node.internal.type === `MarkdownRemark`) {
    if (typeof node.frontmatter.slug !== "undefined") {
      const dirname = getNode(node.parent).relativeDirectory
      createNodeField({
        node,
        name: "slug",
        value: `/${dirname}/${node.frontmatter.slug}`,
      })
    } else {
      const value = createFilePath({ node, getNode })
      createNodeField({
        node,
        name: "slug",
        value,
      })
    }
  }
}

module.exports = onCreateNode
