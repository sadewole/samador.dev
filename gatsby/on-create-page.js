const path = require(`path`)
const createBlogs = require("./pagination/create-blogs")

const createPages = async ({ graphql, actions, reporter }) => {
  const { createPage } = actions

  // Posts and pages from markdown
  const result = await graphql(`
    {
      allMarkdownRemark(filter: { frontmatter: { draft: { ne: true } } }) {
        nodes {
          frontmatter {
            slug
            layout
            title
          }
        }
      }
    }
  `)

  if (result.errors) {
    reporter.panicOnBuild(`There was an error loading this page`, result.errors)
    return
  }

  const { nodes } = result.data.allMarkdownRemark

  nodes.forEach(node => {
    if (node.frontmatter.layout === "page") {
      createPage({
        path: node.frontmatter.slug,
        component: path.resolve("./src/templates/page-template.js"),
        context: { slug: node.frontmatter.slug },
      })
    } else if (node.frontmatter.layout === "blog") {
      createPage({
        path: `/blogs/${node.frontmatter.slug}`,
        component: path.resolve("./src/templates/blog-template.js"),
        context: { slug: node.frontmatter.slug },
      })
    }
  })

  await createBlogs(graphql, actions)
}

module.exports = createPages
