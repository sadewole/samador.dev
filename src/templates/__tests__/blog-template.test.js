import React from "react"
import renderer from "react-test-renderer"
import { useStaticQuery, StaticQuery } from "gatsby"

import BlogTemplate from "../blog-template"
import siteMetadata from "../../../jest/__fixtures__/site-metadata"

describe("Blog Template", () => {
  beforeEach(() => {
    StaticQuery.mockImplementationOnce(
      ({ render }) => render(siteMetadata),
      useStaticQuery.mockReturnValue(siteMetadata)
    )
  })

  const props = {
    data: {
      markdownRemark: {
        id: "1",
        html: "<p>test</p>",
        frontmatter: {
          date: "test-date",
          tags: ["test tag"],
          title: "test",
        },
      },
    },
  }
  it("render correctly", () => {
    const tree = renderer.create(<BlogTemplate {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
