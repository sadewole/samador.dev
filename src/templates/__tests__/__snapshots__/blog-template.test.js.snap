// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blog Template render correctly 1`] = `
<div
  className="py-10 sm:px-10 bg-gray-700 text-gray-300 dark:bg-gray-900 m-0 min-h-screen overflow-x-hidden"
>
  <div
    className="max-w-6xl xl:w-10/12  m-auto"
  >
    <main
      className="max-w-3xl m-auto layout-body"
    >
      <nav
        className="mb-5 flex items-center justify-between"
      >
        <h3
          data-testid="blog-nav"
        >
          <a
            href="/blogs"
          >
            Test name
            's blog
          </a>
        </h3>
        <button
          aria-hidden="true"
          className={null}
          onClick={[Function]}
          style={
            Object {
              "MozAppearance": "none",
              "WebkitAppearance": "none",
              "appearance": "none",
              "backgroundColor": "transparent",
              "border": "none",
              "cursor": "pointer",
              "height": "37.599999999999994px",
              "overflow": "hidden",
              "padding": 0,
              "width": "80px",
            }
          }
        >
          <div
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "height": "132px",
                "justifyContent": "center",
                "marginLeft": "-25.6px",
                "marginTop": "-47.599999999999994px",
                "width": "132px",
              }
            }
          >
            <div
              aria-label="animation"
              role="button"
              style={
                Object {
                  "height": "100%",
                  "margin": "0 auto",
                  "outline": "none",
                  "overflow": "hidden",
                  "width": "100%",
                }
              }
              tabIndex={0}
              title=""
            />
          </div>
        </button>
      </nav>
      <article
        className="mb-15"
      >
        <header
          className="mb-10"
        >
          <h1>
            test
          </h1>
          <time
            className="italic"
            data-testid="publish"
            dateTime="test-date"
          >
            Published on 
            test-date
          </time>
        </header>
        <section>
          <div
            dangerouslySetInnerHTML={
              Object {
                "__html": "<p>test</p>",
              }
            }
          />
        </section>
      </article>
      <footer>
        <div>
          <ul
            className="tags"
          >
            <li
              className="text-white hover:border-gray-600 border-white border p-3 mr-2 text-center inline-flex items-center justify-center h-10 mb-6 shadow-lg rounded-full hover:bg-white hover:text-gray-600 cursor-pointer"
            >
              test tag
            </li>
          </ul>
        </div>
        <hr />
        <div
          className="my-5 flex items-center"
        >
          <img
            alt="Test name"
            className="rounded mr-3"
            height="60"
            src="/test.jpg"
            width="60"
          />
          <div>
            <p>
              Written by
               
              <a
                className=""
                href="https://www.twitter.com/#"
                rel="noopener noreferrer"
                target="_blank"
              >
                <strong>
                  Test name
                </strong>
              </a>
               
            </p>
            <p
              className="text-sm font-light leading-relaxed mt-0 mb-4 text-gray-300"
            >
              Test bio
              .
            </p>
          </div>
        </div>
        <a
          href="/blogs"
          rel="prev"
        >
          ‚Üê Read more articles
        </a>
      </footer>
    </main>
  </div>
</div>
`;
